tests/unit/backend/chalice/api_server/test_v1_dataset.py

```diff
@@ -98,3 +98,107 @@ def test__minimal_status__ok(self):
             processing_status_updater(processing_status_id, processing_status)
             response = self.app.get(test_url.url, headers=headers)
             self.assertEqual(json.loads(response.body)["upload_status"], status.name)
+
+    def test__cancel_dataset_download__ok(self):
+        # Test pre upload
+        collection = self.generate_collection(visibility=CollectionVisibility.PRIVATE.name)
+        processing_status = {"upload_status": UploadStatus.WAITING, "upload_progress": 0.0}
+        dataset = self.generate_dataset(collection=collection, processing_status=processing_status)
+        test_url = f"/dp/v1/datasets/{dataset.id}"
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url, headers=headers)
+
+        self.assertEqual(response.status_code, 202)
+
+        # Test while uploading
+        processing_status = {"upload_status": UploadStatus.UPLOADING, "upload_progress": 10.0}
+        dataset = self.generate_dataset(collection=collection, processing_status=processing_status)
+        test_url = f"/dp/v1/datasets/{dataset.id}"
+
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url, headers=headers)
+        self.assertEqual(response.status_code, 202)
+
+    def test__cancel_dataset_download__dataset_does_not_exist(self):
+        test_url = "/dp/v1/datasets/missing_dataset_id"
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test__delete_uploaded_dataset__ok(self):
+        collection = self.generate_collection(visibility=CollectionVisibility.PRIVATE.name)
+        processing_status = {"upload_status": UploadStatus.UPLOADED, "upload_progress": 0.0}
+        dataset = self.generate_dataset(collection=collection, processing_status=processing_status)
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+
+        # check dataset in collection
+        collection_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        collection_url.add(query_params=dict(visibility=CollectionVisibility.PRIVATE.name))
+        response = self.app.get(collection_url.url, headers=headers)
+        response.raise_for_status()
+        body = json.loads(response.body)
+        dataset_ids = [dataset["id"] for dataset in body["datasets"]]
+        self.assertIn(dataset.id, dataset_ids)
+
+        # delete dataset
+        test_url = f"/dp/v1/datasets/{dataset.id}"
+        response = self.app.delete(test_url, headers=headers)
```

Comment:
add a test to call it again and make sure it returns 403.

Reply:
re the ticket it should return a 202 I think
`DELETE /dp/v1/dataset/{dataset_uuid} always returns 202 even if the dataset does not exists or has been tombstoned.`